### PR TITLE
ATF Script Runner

### PR DESCRIPTION
**Pull request by dboltovskyi**
ATF is a great tool for both developers and testers that help test any change in SDL. But it lacks a powerful test script runner.

### CODE REVIEW

**Comment by AStasiuk** [comment]
@theresalech proposal is ready for review

File: proposals/NNNN-atf-script-runner.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  - Possibility to run test scripts in parallel (in separate threads)
+  - Ability to create reports with logs collected in a format used by continuous integration system
+
+## Potential downsides
+
+N/A
+
+## Impact on existing code
+
+No impact on existing code is observed.
```
Comment (theresalech) [inline proposals/NNNN-atf-script-runner.md]: The ATF code would be impacted, correct?  Can you please provide some of that information here?

File: proposals/NNNN-atf-script-runner.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Motivation
+
+Currently ATF is unable to:
+
+  - run multiple test scripts
+  - run test script in parallel
+  - create test reports in standard format
+
+## Proposed solution
```
Comment (theresalech) [inline proposals/NNNN-atf-script-runner.md]: Please provide more information on how this proposed solution would be implemented.

Comment (dboltovskyi) [reply proposals/NNNN-atf-script-runner.md]: @theresalech Proposal has been updated: added diagram and made some corrections.
Regarding the implementation - it's not too much can be added. Script runner is just a command line tool that can be developed using any programming language, e.g. C++, Python etc.

File: proposals/NNNN-atf-script-runner.md (hunk 1 above)
Comment (dboltovskyi) [reply proposals/NNNN-atf-script-runner.md]: @theresalech Since Script Runner is completely new tool, existing code of ATF won't be affected.

**Pull request merged**